From 3afb71b697b85c129a8e38a1e2a7f6e9a4594f06 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 24 Jul 2017 16:56:56 +0800
Subject: [PATCH 2215/5242] MLK-16075-20 drm/imx: dpu: plane: Use a better way
 to calculate base address

commit  f09e4948cb2d92aab00d9bcc9d9c223447352147 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch uses the helper drm_format_plane_cpp() to calculate base address
so that we can calculate correctly for the YUV pixel formats as well.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dpu/dpu-plane.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/imx/dpu/dpu-plane.c b/drivers/gpu/drm/imx/dpu/dpu-plane.c
index 4e553f6..1d2d68c 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-plane.c
+++ b/drivers/gpu/drm/imx/dpu/dpu-plane.c
@@ -115,13 +115,14 @@ static void dpu_drm_atomic_plane_destroy_state(struct drm_plane *plane,
 {
 	struct drm_framebuffer *fb = state->fb;
 	struct drm_gem_cma_object *cma_obj;
+	unsigned int x = state->src_x >> 16;
+	unsigned int y = state->src_y >> 16;
 
 	cma_obj = drm_fb_cma_get_gem_obj(fb, 0);
 	BUG_ON(!cma_obj);
 
-	return cma_obj->paddr + fb->offsets[0] +
-	       fb->pitches[0] * (state->src_y >> 16) +
-	       fb->format->cpp[0] * (state->src_x >> 16);
+	return cma_obj->paddr + fb->offsets[0] + fb->pitches[0] * y +
+	       drm_format_plane_cpp(fb->format->format, 0) * x;
 }
 
 static int dpu_plane_atomic_check(struct drm_plane *plane,
-- 
1.7.9.5

